django/db/models/constraints.py

```diff
@@ -122,4 +133,23 @@ def deconstruct(self):
         kwargs['fields'] = self.fields
         if self.condition:
             kwargs['condition'] = self.condition
+        if self.defer:
+            kwargs['defer'] = self.defer
         return path, args, kwargs
+
+
+DEFERRED = 'DEFERRED'
+IMMEDIATE = 'IMMEDIATE'
+
+
+def set_constraints(using=None, **kwargs):
```

Comment:
Should an exception be raised when this is called outside of an exception?

Reply:
Outside of a transaction you mean? I think so, yes. Is there a standard way to detect that?

Reply:
Sorry yes I meant `transaction` 🤦‍♂. I think there's `connections[using].in_atomic_block` or something like that.